language/cc/resolve.go

```diff
@@ -58,11 +59,22 @@ func (*ccLanguage) Imports(c *config.Config, r *rule.Rule, f *rule.File) []resol
 		if includePrefix != "" {
 			includePrefix = path.Clean(includePrefix)
 		}
-		imports = make([]resolve.ImportSpec, len(hdrs))
-		for i, hdr := range hdrs {
+		includes := r.AttrStrings("includes")
+		for i, includeDir := range includes {
+			includes[i] = filepath.FromSlash(path.Clean(includeDir))
+		}
+		imports = make([]resolve.ImportSpec, 0, len(hdrs)*(1+len(includes)))
+		for _, hdr := range hdrs {
 			hdrRel := path.Join(f.Pkg, hdr)
 			inc := transformIncludePath(f.Pkg, stripIncludePrefix, includePrefix, hdrRel)
-			imports[i] = resolve.ImportSpec{Lang: languageName, Imp: inc}
+			// Index path directly as it is
+			imports = append(imports, resolve.ImportSpec{Lang: languageName, Imp: inc})
+			// Index each matching relative include paths
+			for _, include := range includes {
+				if relInclude, err := filepath.Rel(include, inc); err == nil {
```

Comment:
Why is `filepath.Rel` needed? I think we can compute the `-isystem` path added to the command line (`path.Join(c.Rel, include)`), then use that to figure out whether that's a prefix of `inc` and what the rest of it is.

Reply:
I've replaced it with suggested approach. I think it doesn't really change the behaviour,  but it allowed to me introduce 1 missing logic part - includes allows for `.` which results in resolving include relative to current (BUILD) directory instead of project root.